src/rules/no-multiple-animated-style-usages.ts

```diff
@@ -62,20 +76,8 @@ export default createRule<Options, MessageIds>({
         }
         definedAnimatedStyles.add(definition);
       },
-      "JSXAttribute[name.name='style'] Identifier": (
-        node: TSESTree.Identifier
-      ) => {
-        const variable = getVariableInScope(node.name, context.getScope());
-        const definition = variable?.defs?.[0];
-        if (!definition) {
-          return;
-        }
-        if (!definedAnimatedStyles.has(definition)) {
-          return;
-        }
-        const existingIdentifiers = styleDefinitions.get(definition) ?? [];
-        styleDefinitions.set(definition, [...existingIdentifiers, node]);
-      },
+      "JSXAttribute[name.name='style'] > JSXExpressionContainer > Identifier": checkIdentifier,
+      "JSXAttribute[name.name='style'] > JSXExpressionContainer > ArrayExpression > Identifier": checkIdentifier,
```

Comment:
I switched to a stricter "opt-in" approach with the selectors

So this could miss "false negatives" eg:
```
<>
  <Animated.View style={style} />
  <Animated.View style={[[style]]} />
```
But I think this catches the common cases, probably this is the right tradeoff to avoid false positives?

Reply:
What kind of opt-in and tradeoffs are you referring to?

Another interesting approach would be to check if the variable is of type AnimatedStyleProp and has more than one ref, that could catch really anything (even passing down the style as props multiple times etc). What do you think? too harsh?

Reply:
> What kind of opt-in and tradeoffs are you referring to?

By "opt-in" I mean that with these strict "immediate child" (`>`) selectors we are only explicitly "allowing" certain exact structures of AST elements to be flagged (rather than the previous "descendant" selector approach where by default all identifiers inside JSX attribute values may get flagged, and then we'd have to "opt out"/"disallow" specific false positive cases)

And so the tradeoffs seem to broadly be "catching more true positives" vs "avoiding more/unanticipated false positives"

> Another interesting approach would be to check if the variable is of type AnimatedStyleProp and has more than one ref, that could catch really anything (even passing down the style as props multiple times etc). What do you think? too harsh?

No I think this is a good idea and yes more robust. I initially thought of checking the type for `AnimatedStyleProp` but then saw that animated styles were reporting their types as simple object types (eg `{opacity: number}`). Even though I don't think at runtime that is truly their type?

But I can "decouple" your suggestions here. Even if we don't check for `AnimatedStyleProp` type, we can still use multiple-ref checking. I looked more closely at the scope variable/definition/reference types and by storing a _variable_ when we see a `useAnimatedStyle()` definition, then when we see an identifier inside a JSX attribute we can check if that identifier is one of the references of those known animated style variables (and keep a count of those references)

So I updated to use this approach and this is a win-win: keep the simpler selector logic while avoiding false positives like the member expression case. Thanks for the suggestion!

> has more than one ref ... too harsh?

This is an interesting question. Simple "has more than one ref" would false-positive flag the one test case where there is a `console.log({style})`. But it would also catch other true-positives that the current heuristic of "multiple refs that live inside a JSX attribute value" doesn't, eg I included a commented-out test case:
```
const styles = [style]
return (
  <>
    <Animated.View style={style} />
    <Animated.View style={styles} />
  </>
)
```
What do you think?

Reply:
tl;dr I updated to use multiple ref checking but still using "has more than one ref inside a JSX attribute value" rather than "has more than one ref"